File: src/main/java/arvem/skykid/registries/RegisterSounds.java
package arvem.skykid.registries;

import arvem.skykid.Skykid;
import net.minecraft.registry.Registry;
import net.minecraft.registry.Registries;
import net.minecraft.sound.SoundEvent;
import net.minecraft.util.Identifier;

public class RegisterSounds {
    // Identifiers
    private static final Identifier CAPE_FLAP_ID       = Identifier.of(Skykid.MODID, "player.flap");
    private static final Identifier CAPE_BOOST_ID      = Identifier.of(Skykid.MODID, "player.boost");
    private static final Identifier CAPE_WHOOSH_ID     = Identifier.of(Skykid.MODID, "player.whoosh");
    private static final Identifier SHOUT_DEFAULT_ID   = Identifier.of(Skykid.MODID, "player.default");
    private static final Identifier SHOUT_BIRD_ID      = Identifier.of(Skykid.MODID, "player.bird");
    private static final Identifier SHOUT_CRAB_ID      = Identifier.of(Skykid.MODID, "player.crab");
    private static final Identifier SHOUT_GHOST_ID     = Identifier.of(Skykid.MODID, "player.ghost");
    private static final Identifier SHOUT_WHALE_ID     = Identifier.of(Skykid.MODID, "player.whale");
    private static final Identifier SHOUT_JELLY_ID     = Identifier.of(Skykid.MODID, "player.jelly");
    private static final Identifier SHOUT_MANTA_ID     = Identifier.of(Skykid.MODID, "player.manta");
    private static final Identifier BEAT_SLOW_ID       = Identifier.of(Skykid.MODID, "player.beatslow");
    private static final Identifier BEAT_MID_ID        = Identifier.of(Skykid.MODID, "player.beatmid");
    private static final Identifier BEAT_FAST_ID       = Identifier.of(Skykid.MODID, "player.beatfast");

    // SoundEvent instances
    public static final SoundEvent CAPE_FLAP_EVENT     = SoundEvent.of(CAPE_FLAP_ID);
    public static final SoundEvent CAPE_BOOST_EVENT    = SoundEvent.of(CAPE_BOOST_ID);
    public static final SoundEvent CAPE_WHOOSH_EVENT   = SoundEvent.of(CAPE_WHOOSH_ID);
    public static final SoundEvent SHOUT_DEFAULT_EVENT = SoundEvent.of(SHOUT_DEFAULT_ID);
    public static final SoundEvent SHOUT_BIRD_EVENT    = SoundEvent.of(SHOUT_BIRD_ID);
    public static final SoundEvent SHOUT_CRAB_EVENT    = SoundEvent.of(SHOUT_CRAB_ID);
    public static final SoundEvent SHOUT_GHOST_EVENT   = SoundEvent.of(SHOUT_GHOST_ID);
    public static final SoundEvent SHOUT_WHALE_EVENT   = SoundEvent.of(SHOUT_WHALE_ID);
    public static final SoundEvent SHOUT_JELLY_EVENT   = SoundEvent.of(SHOUT_JELLY_ID);
    public static final SoundEvent SHOUT_MANTA_EVENT   = SoundEvent.of(SHOUT_MANTA_ID);
    public static final SoundEvent BEAT_SLOW_EVENT     = SoundEvent.of(BEAT_SLOW_ID);
    public static final SoundEvent BEAT_MID_EVENT      = SoundEvent.of(BEAT_MID_ID);
    public static final SoundEvent BEAT_FAST_EVENT     = SoundEvent.of(BEAT_FAST_ID);

    public static void init() {
        // Single-line registrations via helper
        register(CAPE_FLAP_ID,     CAPE_FLAP_EVENT);
        register(CAPE_BOOST_ID,    CAPE_BOOST_EVENT);
        register(CAPE_WHOOSH_ID,   CAPE_WHOOSH_EVENT);
        register(SHOUT_DEFAULT_ID, SHOUT_DEFAULT_EVENT);
        register(SHOUT_BIRD_ID,    SHOUT_BIRD_EVENT);
        register(SHOUT_CRAB_ID,    SHOUT_CRAB_EVENT);
        register(SHOUT_GHOST_ID,   SHOUT_GHOST_EVENT);
        register(SHOUT_WHALE_ID,   SHOUT_WHALE_EVENT);
        register(SHOUT_JELLY_ID,   SHOUT_JELLY_EVENT);
        register(SHOUT_MANTA_ID,   SHOUT_MANTA_EVENT);
        register(BEAT_SLOW_ID,     BEAT_SLOW_EVENT);
        register(BEAT_MID_ID,      BEAT_MID_EVENT);
        register(BEAT_FAST_ID,     BEAT_FAST_EVENT);
    }

    private static void register(Identifier id, SoundEvent event) {
        Registry.register(Registries.SOUND_EVENT, id, event);
    }
}

File: src/main/kotlin/arvem/skykid/registries/RegisterSounds.kt
package arvem.skykid.registries

import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.sound.SoundEvent
import net.minecraft.util.Identifier

object RegisterSounds {
    private val CAPE_FLAP = Identifier("skykid:player.flap")
    private var CAPE_FLAP_EVENT: SoundEvent = SoundEvent.of(CAPE_FLAP)
    private val CAPE_BOOST = Identifier("skykid:player.boost")
    private var CAPE_BOOST_EVENT: SoundEvent = SoundEvent.of(CAPE_BOOST)
    private val CAPE_WHOOSH = Identifier("skykid:player.whoosh")
    private var CAPE_WHOOSH_EVENT: SoundEvent = SoundEvent.of(CAPE_WHOOSH)
    private val SHOUT_DEFAULT = Identifier("skykid:player.default")
    private var SHOUT_DEFAULT_EVENT: SoundEvent = SoundEvent.of(SHOUT_DEFAULT)
    private val SHOUT_BIRD = Identifier("skykid:player.bird")
    private var SHOUT_BIRD_EVENT: SoundEvent = SoundEvent.of(SHOUT_BIRD)
    private val SHOUT_CRAB = Identifier("skykid:player.crab")
    private var SHOUT_CRAB_EVENT: SoundEvent = SoundEvent.of(SHOUT_CRAB)
    private val SHOUT_GHOST = Identifier("skykid:player.ghost")
    private var SHOUT_GHOST_EVENT: SoundEvent = SoundEvent.of(SHOUT_GHOST)
    private val SHOUT_WHALE = Identifier("skykid:player.whale")
    private var SHOUT_WHALE_EVENT: SoundEvent = SoundEvent.of(SHOUT_WHALE)
    private val SHOUT_JELLY = Identifier("skykid:player.jelly")
    private var SHOUT_JELLY_EVENT: SoundEvent = SoundEvent.of(SHOUT_JELLY)
    private val SHOUT_MANTA = Identifier("skykid:player.manta")
    private var SHOUT_MANTA_EVENT: SoundEvent = SoundEvent.of(SHOUT_MANTA)
    private val BEAT_SLOW = Identifier("skykid:player.beatslow")
    private var BEAT_SLOW_EVENT: SoundEvent = SoundEvent.of(BEAT_SLOW)
    private val BEAT_MID = Identifier("skykid:player.beatmid")
    private var BEAT_MID_EVENT: SoundEvent = SoundEvent.of(BEAT_MID)
    private val BEAT_FAST = Identifier("skykid:player.beatfast")
    private var BEAT_FAST_EVENT: SoundEvent = SoundEvent.of(BEAT_FAST)
    fun init() {
        Registry.register(Registries.SOUND_EVENT, CAPE_FLAP, CAPE_FLAP_EVENT)
        Registry.register(Registries.SOUND_EVENT, CAPE_BOOST, CAPE_BOOST_EVENT)
        Registry.register(Registries.SOUND_EVENT, CAPE_WHOOSH, CAPE_WHOOSH_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_DEFAULT, SHOUT_DEFAULT_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_BIRD, SHOUT_BIRD_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_CRAB, SHOUT_CRAB_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_GHOST, SHOUT_GHOST_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_WHALE, SHOUT_WHALE_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_JELLY, SHOUT_JELLY_EVENT)
        Registry.register(Registries.SOUND_EVENT, SHOUT_MANTA, SHOUT_MANTA_EVENT)
        Registry.register(Registries.SOUND_EVENT, BEAT_SLOW, BEAT_SLOW_EVENT)
        Registry.register(Registries.SOUND_EVENT, BEAT_MID, BEAT_MID_EVENT)
        Registry.register(Registries.SOUND_EVENT, BEAT_FAST, BEAT_FAST_EVENT)
    }
}
